pyproject.toml

```
@@ -45,12 +45,14 @@ Repository = "https://github.com/facebookresearch/optimizers.git"
 "Bug Tracker" = "https://github.com/facebookresearch/optimizers/issues"
 
 [tool.setuptools]
-packages = ["distributed_shampoo"]
 py-modules = [
     "matrix_functions",
     "matrix_functions_types",
     "optimizer_modules",
 ]
 
+[tool.setuptools.packages.find]
+include = ["distributed_shampoo*"]
```

I think we want to do this instead, to avoid packaging the tests and examples:
```suggestion
exclude = ["*tests", "*examples"]
```
But I'm not sure how this could be related to the failing GPU tests.

---

Unfortunately, this does not work. First, without `include`, `setuptools` will install too much things (`build/lib`). Second, `exclude` seems to have no effect.

---

I see. Not sure why `exclude` has no effect though? Do you know how we can exclude `*tests` and `*examples`?

---

There seems to be a lot of long lasting issues and confusion regarding `exclude` ... (e.g. https://github.com/pypa/setuptools/discussions/3346, https://github.com/pypa/setuptools/issues/3260). I would say, the easiest is likely to move these test directories outside of the module (they should not be there anyway).

---

Agreed, that makes sense to me. Thanks a lot for your fix and help with debugging!